pkg/webhook/network/mutator.go

```diff
@@ -61,9 +61,21 @@ func (m *mutator) Mutate(ctx context.Context, obj runtime.Object) error {
 		return nil
 	}
 
-	network, ok := obj.(*extensionsv1alpha1.Network)
+	newNetwork, ok := newObj.(*extensionsv1alpha1.Network)
 	if !ok {
 		return fmt.Errorf("could not mutate, object is not of type \"Network\"")
 	}
-	return m.mutateFunc(network)
+
+	var oldNetwork *extensionsv1alpha1.Network
+
+	if oldObj == nil {
+		old, ok := oldObj.(*extensionsv1alpha1.Network)
+		if !ok {
+			return fmt.Errorf("could not mutate, old object is not of type \"Network\"")
```

Comment:
It is up the the called to return "cannot mutate".
```suggestion
			return fmt.Errorf("could not cast old object to extensionsv1alpha1.Network")
```